CMakeLists.txt

```diff
@@ -1,9 +1,10 @@
 #=========================================================
-CMAKE_MINIMUM_REQUIRED(VERSION 3.12.0 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.0 FATAL_ERROR)
```

Comment:
If this PR/branch is to prepare up-streaming this change, I think we shouldn't change anything in CMakeLists, otherwise I think it won't be accepted.

Anyway, this changes looks like mistake, if you need c++14 for ``std::make_unique`` do not use it. Instead write own function for this.

Reply:
I changed it back to 3.12. I also removed three lines which I added (42-44).